pkg/controllers/node/emptiness.go

```diff
@@ -0,0 +1,91 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha3"
+	"github.com/awslabs/karpenter/pkg/utils/functional"
+	"github.com/awslabs/karpenter/pkg/utils/node"
+	"github.com/awslabs/karpenter/pkg/utils/pod"
+	"github.com/awslabs/karpenter/pkg/utils/ptr"
+	v1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/logging"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Emptiness is a subreconciler that terminates nodes that are empty after a ttl
+type Emptiness struct {
+	kubeClient client.Client
+}
+
+// Reconcile reconciles the node
+func (r *Emptiness) Reconcile(ctx context.Context, provisioner *v1alpha3.Provisioner, n *v1.Node) (reconcile.Result, error) {
+	// 1. Ignore node if not applicable
```

Comment:
Can you change this line to the following to match the expiration controller?
```
	// 1. Ignore if TTLSecondsAfterEmpty isn't defined 
```

Reply:
I'd rather do the reverse and use "not applicable" since it's less likely to get stale.

Reply:
That sounds good to me. I'd like to keep the comments roughly the same irrespective of the content.